crates/rune/src/ast/item.rs

```diff
@@ -47,12 +67,21 @@ impl Item {
         })
     }
 
-    /// Parse an item within a nested block
-    pub fn parse_in_nested_block(parser: &mut Parser) -> Result<Self, ParseError> {
-        let attributes: Vec<ast::Attribute> = parser.parse()?;
+    /// Parse an Item attaching the given Attributes
+    pub fn parse_with_attributes(
+        parser: &mut Parser,
+        attributes: Vec<ast::Attribute>,
+    ) -> Result<Self, ParseError> {
         let t = parser.token_peek_eof()?;
 
-        Ok(match t.kind {
+        let kind = if t.kind == ast::Kind::Pub {
+            let t2 = parser.token_peek2_eof()?;
```

Comment:
This probably doesn't work for restricted visibility (e.g. `pub(crate)`)? So we'll need something `parse_with_attributes_and_vis` (or `parse_with_meta`) for the items to do this correctly.

I don't have too much of an issue about it for now, since visibility is not supported anyway. But should be fixed soon:ish.

Reply:
Good catch. Yeah it will definitely fail soon as there is an actual use/test of it.

Makes me wonder if we can come up with general pattern for common "prefixes" of ast structs?